src/main/java/org/opensearch/knn/index/VectorDataType.java

```diff
@@ -0,0 +1,293 @@
+ */
+public enum VectorDataType {
+    BYTE("byte") {
+        /**
+         * @param dimension  Dimension of the vector
+         * @param vectorSimilarityFunction VectorSimilarityFunction for a given spaceType
+         * @return FieldType of type KnnByteVectorField
+         */
+        @Override
+        public FieldType createKnnVectorFieldType(int dimension, VectorSimilarityFunction vectorSimilarityFunction) {
+            return KnnByteVectorField.createFieldType(dimension, vectorSimilarityFunction);
+        }
+
+        /**
+         * @param knnEngine KNNEngine
+         * @return DocValues FieldType of type Binary and with BYTE VectorEncoding
+         */
+        @Override
+        public FieldType buildDocValuesFieldType(KNNEngine knnEngine) {
+            IndexableFieldType indexableFieldType = new IndexableFieldType() {
+                @Override
+                public boolean stored() {
+                    return false;
+                }
+
+                @Override
+                public boolean tokenized() {
+                    return true;
+                }
+
+                @Override
+                public boolean storeTermVectors() {
+                    return false;
+                }
+
+                @Override
+                public boolean storeTermVectorOffsets() {
+                    return false;
+                }
+
+                @Override
+                public boolean storeTermVectorPositions() {
+                    return false;
+                }
+
+                @Override
+                public boolean storeTermVectorPayloads() {
+                    return false;
+                }
+
+                @Override
+                public boolean omitNorms() {
+                    return false;
+                }
+
+                @Override
+                public IndexOptions indexOptions() {
+                    return IndexOptions.NONE;
+                }
+
+                @Override
+                public DocValuesType docValuesType() {
+                    return DocValuesType.NONE;
+                }
+
+                @Override
+                public int pointDimensionCount() {
+                    return 0;
+                }
+
+                @Override
+                public int pointIndexDimensionCount() {
+                    return 0;
+                }
+
+                @Override
+                public int pointNumBytes() {
+                    return 0;
+                }
+
+                @Override
+                public int vectorDimension() {
+                    return 0;
+                }
+
+                @Override
+                public VectorEncoding vectorEncoding() {
+                    return VectorEncoding.BYTE;
+                }
+
+                @Override
+                public VectorSimilarityFunction vectorSimilarityFunction() {
+                    return VectorSimilarityFunction.EUCLIDEAN;
+                }
```

Comment:
shouldn't this be the one which we are taking in the index creation request?

Reply:
Actually that is where I need some confirmation. For lucene when we are [building the docValues FieldType](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java#L73-L79) we are only setting the knn engine and docValues Type as BINARY. I don't think we need to set the vectorSimilarityFunction with the one that is provided in mapping because in lucene to [index docValues](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexingChain.java#L941) they are not using it, just need the docValues type and serialized form of vector.